snapcraft/internal/yaml.py

```diff
@@ -186,13 +190,16 @@ def _compute_part_dependencies(self):
                         found = True
                         break
                 if not found:
-                    wiki_part = self._wiki.get_part(dep)
-                    found = True if wiki_part else False
+                    remote_part = self._remote_parts.get_part(dep)
+                    found = True if remote_part else False
                     if found:
-                        plugin_name = wiki_part.pop('plugin')
+                        plugin_name = remote_part.pop('plugin')
                         part.deps.append(self.load_plugin(
-                            dep, plugin_name, wiki_part))
+                            dep, plugin_name, remote_part))
                         self._part_names.append(dep)
+                    else:
+                        logger.info('Maybe {!r} is defined as a remote part, '
+                                    'run `snapcraft update` to refresh')
```

Comment:
also missing a test. And should this be an error, shouldn't it?
This logic is so weird. I think this else should raise the exception.
And I don't have it clear if in that case, the following statement is needed.
Splitting smaller functions with nice names out of these loops would be nice, but again, maybe for a future refactor.